### DISCUSSION TITLE
Google Analytics

### DISCUSSION THREAD

**Discussion by supermx1** [directus/directus#13171] [Feature Requests]
It would be nice to have google analytics or any other tracking tool embedded with the Directus app or better yet ability to add custom javascript in the header or body section. I know this is a huge ask as it presents some security loopholes, but it would be nice to be able to have a way to have in-depth tracking and analytics of user behavior and more.

**Comment by benhaynes** [comment]
I agree. Not sure if the JS Override value could be sanitized... probably not. Could be an ENV variable to toggle the setting?

**Comment by u12206050** [comment]
In a more general sense, being able to load any 3rd party js script seems impossible due to the security measures in place. We can't even embed a support chat bubble without running into this problem.
So would be very nice to be able to add custom scripts somewhere and have them exempt from the security policies.

**Reply by rijkvanzanten** [reply]
> In a more general sense, being able to load any 3rd party js script seems impossible due to the security measures in place

By default that is, you can tweak any of the content-security-policy headers by using the `CONTENT_SECURITY_POLICY_*` environment variables 👍🏻

**Reply by marcelloromanelli** [reply]
@u12206050 I'm running into similar issues as I'm trying to add a 3rd party library to the `head` section. How did you solve it for intercom? Did you add it just to a specific module?

**Comment by u12206050** [comment]
Thanks for that @rijkvanzanten Turned out I needed quite a few of those. Here are the settings for anyone wanting to use Intercom

```
CONTENT_SECURITY_POLICY_DIRECTIVES__SCRIPT_SRC="'self' 'unsafe-eval' https://widget.intercom.io https://js.intercomcdn.com"
CONTENT_SECURITY_POLICY_DIRECTIVES__FRAME_SRC="https://js.intercomcdn.com"
CONTENT_SECURITY_POLICY_DIRECTIVES__MEDIA_SRC="'self' https://cdn.directus.io https://js.intercomcdn.com"
CONTENT_SECURITY_POLICY_DIRECTIVES__IMG_SRC="'self' data: blob: https://cdn.directus.io https://js.intercomcdn.com https://downloads.intercomcdn.com https://static.intercomassets.com"
CONTENT_SECURITY_POLICY_DIRECTIVES__CONNECT_SRC="'self' https://* wss://*"
```

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
